doc/specs/stdlib_math.md

```diff
@@ -649,3 +649,88 @@ program demo_math_all_close
     
 end program demo_math_all_close
 ```
+
+### `diff`
+
+#### Description
+
+Computes differences between adjacent elements of an array.
+
+#### Syntax
+
+For a rank-1 array
+```fortran
+y = [[stdlib_math(module):diff(interface)]](x [, n, prepend, append])
+```
+and for a rank-2 array
+```fortran
+y = [[stdlib_math(module):diff(interface)]](x [, n, dim, prepend, append])
+```
+
+#### Status
+
+Experimental.
+
+#### Class
+
+Pure function.
+
+#### Arguments
+
+`x`: The array to take a difference of.
+Shall be a `real/integer` and `rank-1/rank-2` array.
+This argument is `intent(in)`.
+
+`n`: The order of finite differencing to use.
+Shall be an `integer` scalar.
+This argument is `intent(in)` and `optional`, which is `1` by default.
```

Comment:
I apologize, now I realize that this wording could be confused for the accuracy order of the finite difference scheme. How about:

```suggestion
`n`: How many times to iteratively calculate the difference.
Shall be an `integer` scalar.
This argument is `intent(in)` and `optional`, and has value of `1` by default.
```

Reply:
Thanks, I feel good, now updated.